# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.h
# Commit: 01c2bc6d727a
# Full Hash: 01c2bc6d727a70be74dbf7babb7f6294824a571a
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 10 - CookieStore API - CHIP implementation shared between Document and CookieStore, r=cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220288
# ==============================================================================

diff -r 2d925ea3da19 -r 01c2bc6d727a netwerk/cookie/CookieCommons.h
--- a/netwerk/cookie/CookieCommons.h	Tue Sep 10 16:31:41 2024 +0000
+++ b/netwerk/cookie/CookieCommons.h	Tue Sep 10 16:31:41 2024 +0000
@@ -115,8 +115,16 @@
   static already_AddRefed<nsICookieJarSettings> GetCookieJarSettings(
       nsIChannel* aChannel);
 
-  static bool ShouldIncludeCrossSiteCookieForDocument(Cookie* aCookie,
-                                                      dom::Document* aDocument);
+  static bool ShouldIncludeCrossSiteCookie(Cookie* aCookie,
+                                           bool aPartitionForeign,
+                                           bool aInPrivateBrowsing,
+                                           bool aUsingStorageAccess);
+
+  static bool ShouldIncludeCrossSiteCookie(int32_t aSameSiteAttr,
+                                           bool aCookiePartitioned,
+                                           bool aPartitionForeign,
+                                           bool aInPrivateBrowsing,
+                                           bool aUsingStorageAccess);
 
   static bool IsSchemeSupported(nsIPrincipal* aPrincipal);
   static bool IsSchemeSupported(nsIURI* aURI);
